pkg/kconfig/kconfig.go

```diff
@@ -326,6 +326,29 @@ func (kp *kconfigParser) parseProperty(prop string) {
 }
 
 func (kp *kconfigParser) includeSource(file string) {
+	// Some includes use environment variables.
+	// This table contains vars to extend before processing.
+	// Add more here if needed.
+
+	envVars := []string{
+		"KCONFIG_EXT_PREFIX",
+		"SRCARCH", // "x86" or "arm64"
```

Comment:
Running with this change gives me stack overflow:

```
kernel release v5.15
generating android-5.15...
runtime: goroutine stack exceeds 1000000000-byte limit
runtime: sp=0xc465290378 stack=[0xc465290000, 0xc485290000]
fatal error: stack overflow

goroutine 131 [running]:
strings.Index({0xc1b927fcf8, 0x17}, {0xc465290518, 0xa})
	strings/strings.go:1024 +0x6f2 fp=0xc465290388 sp=0xc465290380 pc=0x4baa32
strings.Count({0xc1b927fcf8, 0x1}, {0xc465290518, 0xa})
	strings/strings.go:51 +0xb8 fp=0xc4652903d0 sp=0xc465290388 pc=0x4b7198
strings.Replace({0xc1b927fcf8, 0x5dcd94}, {0xc465290518, 0xa}, {0x0, 0x0}, 0xffffffffffffffff)
	strings/strings.go:930 +0xcc fp=0xc465290498 sp=0xc4652903d0 pc=0x4b9a6c
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).includeSource(0xc48528fbe0, {0xc1b927fcf8, 0xc1b9e7b208})
	pkg/kconfig/kconfig.go:347 +0x1d2 fp=0xc4652905f8 sp=0xc465290498 pc=0x4f3fd2
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseMenu(0xc48528fbe0, {0xc1b9e7b208, 0xc36f1fd2c0})
	pkg/kconfig/kconfig.go:209 +0x4b4 fp=0xc465290690 sp=0xc4652905f8 pc=0x4f2f94
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseLine(0xc48528fbe0)
	pkg/kconfig/kconfig.go:199 +0x24e fp=0xc4652906d0 sp=0xc465290690 pc=0x4f290e
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseFile(0xc48528fbe0)
	pkg/kconfig/kconfig.go:164 +0x3b fp=0xc4652906f0 sp=0xc4652906d0 pc=0x4f25bb
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).includeSource(0xc48528fbe0, {0xc1b927fce0, 0xc1b9e7b1d0})
	pkg/kconfig/kconfig.go:362 +0x45d fp=0xc465290850 sp=0xc4652906f0 pc=0x4f425d
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseMenu(0xc48528fbe0, {0xc1b9e7b1d0, 0xc36f1fd140})
	pkg/kconfig/kconfig.go:209 +0x4b4 fp=0xc4652908e8 sp=0xc465290850 pc=0x4f2f94
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseLine(0xc48528fbe0)
	pkg/kconfig/kconfig.go:199 +0x24e fp=0xc465290928 sp=0xc4652908e8 pc=0x4f290e
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseFile(0xc48528fbe0)
	pkg/kconfig/kconfig.go:164 +0x3b fp=0xc465290948 sp=0xc465290928 pc=0x4f25bb
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).includeSource(0xc48528fbe0, {0xc1b927fcc8, 0xc1b9e7b198})
	pkg/kconfig/kconfig.go:362 +0x45d fp=0xc465290aa8 sp=0xc465290948 pc=0x4f425d
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseMenu(0xc48528fbe0, {0xc1b9e7b198, 0xc36f1fcfc0})
	pkg/kconfig/kconfig.go:209 +0x4b4 fp=0xc465290b40 sp=0xc465290aa8 pc=0x4f2f94
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseLine(0xc48528fbe0)
	pkg/kconfig/kconfig.go:199 +0x24e fp=0xc465290b80 sp=0xc465290b40 pc=0x4f290e
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseFile(0xc48528fbe0)
	pkg/kconfig/kconfig.go:164 +0x3b fp=0xc465290ba0 sp=0xc465290b80 pc=0x4f25bb
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).includeSource(0xc48528fbe0, {0xc1b927fcb0, 0xc1b9e7b160})
	pkg/kconfig/kconfig.go:362 +0x45d fp=0xc465290d00 sp=0xc465290ba0 pc=0x4f425d
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseMenu(0xc48528fbe0, {0xc1b9e7b160, 0xc36f1fce40})
	pkg/kconfig/kconfig.go:209 +0x4b4 fp=0xc465290d98 sp=0xc465290d00 pc=0x4f2f94
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseLine(0xc48528fbe0)
	pkg/kconfig/kconfig.go:199 +0x24e fp=0xc465290dd8 sp=0xc465290d98 pc=0x4f290e
github.com/google/syzkaller/pkg/kconfig.(*kconfigParser).parseFile(0xc48528fbe0)
...
```

We shouldn't use actual env vars here. The build should be hermetic. That may be the reason why I am getting the crash while it works for you.

We already replace `SRCARCH` here:
https://github.com/google/syzkaller/blob/267e3bb1576b2f9fa97ae49305aaaa80768ba385/pkg/kconfig/kconfig.go#L417
We need to replace `KCONFIG_EXT_PREFIX` in that function as well (strings may be used not only for includes).

Reply:
That is changed in the new version. @dvyukov please give it a try. Thanks!